security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,34 @@
+package crypter
+
+import (
+	"errors"
+)
+
+// counter is a 64-bit counter.
+type counter struct {
+	value         uint64
+	hasOverflowed bool
+}
+
+func newCounter(value uint64) counter {
```

Comment:
In normal cases, will the counter be started from a non-zero value?

Reply:
I think in normal cases, the counter will start from 0. I'm not sure about any case which starts from a non-zero value right now but I think @matthewstevenson88 had some thoughts on this.

Reply:
In TLS 1.3, the counter will always start at zero (unless we have to increment it by a few because of issues with session tickets, but that seems unlikely given the new resumption design). However, in TLS 1.2, the initial sequence number may be nonzero. This is why we should add this plumbing now instead of hardcoding in TLS 1.3 assumptions. WDYT?